corehq/apps/hqwebapp/tests/utils/test_translation.py

```diff
@@ -0,0 +1,78 @@
+from contextlib import ContextDecorator
+from unittest.mock import MagicMock, patch
+from django.test import SimpleTestCase
+
+from django.utils import translation
+from django.utils.translation import ugettext_lazy
+from django.utils.translation.trans_real import translation as get_translations
+
+from ...utils.translation import mark_safe_lazy, format_html_lazy
+
+CUSTOM_LANGUAGE = 'custom'
+
+
+class custom_translations(ContextDecorator):
+    """
+    A decorator/context manager to provide runtime translations for the 'custom' language
+    """
+    def __init__(self, translation_mapping):
+        self.translation_mapping = translation_mapping
+
+    def __enter__(self):
+        translations = get_translations(CUSTOM_LANGUAGE)
+        old_gettext = translations.gettext
+
+        def lookup(id):
+            return self.translation_mapping.get(id) or old_gettext(id)
+
+        self.patcher = patch.object(translations, 'gettext', MagicMock(side_effect=lookup))
```

Comment:
Why does this need to use `MagicMock`? Would this work?
```suggestion
        self.patcher = patch.object(translations, 'gettext', lookup)
```